### ISSUE TITLE
bug: Wrong executed amounts in competition data

### ISSUE THREAD

**Issue by fhenneke** [cowprotocol/services#2150]
### Problem
It seems that the data on the buy and sell amounts of orders added in #1975 is not accurate.
This is only implemented on staging if I understand correctly.

For example, the transaction with hash `0x2548006a1bf02717d4e0a20bd6f0db7e8762d146161135f2d22f857eb726740d` gives executed amounts
![image](https://github.com/cowprotocol/services/assets/5250107/5d1a2d2a-d72e-431b-8127-16b6b5fe8c6c)
but the sell amount on etherscan
![image](https://github.com/cowprotocol/services/assets/5250107/d695c09f-aa7e-45f7-9425-805bd1adadc7)
and our order book endpoint
![image](https://github.com/cowprotocol/services/assets/5250107/eef9c769-4f7d-4c6c-9916-4d36df337426)
are different.

### Impact
Having the correct data in the competition endpoint is required for EBBO tests.

### Expected behaviour
I expect that the "sellAmount" and "buyAmount" in the competition endpoint to be the amounts send and received by a user in that settlement.

**Comment by fleupold** [comment]
It looks like the amount shown there is the sellAmount + solver fee (which doesn't matter here because it's a "market" order where the fee is specified by the order itself: [log](https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/discover#/doc/ea511870-d9b3-11ed-a9d0-a17451f01cc1/cowlogs-staging-2023.12.08?id=T4ZjS4wB3Q9lqTjuFLAA)

This seems to be because the driver assumes the fee is correctly solver specified if it's present on the response of the solver engine:

https://github.com/cowprotocol/services/blob/03ed19bb7ff7fb9b31fb943f3029372680ee1579/crates/driver/src/infra/solver/dto/solution.rs#L41-L49


We can either check there is the order has a solver defined fee, or fix it in the solver engines to not set this value for order types that don't expect it. The more defensive thing imo would be to change it in the driver.

**Comment by fleupold** [comment]
Actually I think my assessment from yesterday was incorrect, we do already reject settlements from solvers that set a fee on orders they aren't supposed to. The discrepancy between the amount you see and `executedSellAmount` in the API is the difference between the subsidized and unsubsidized fee amount. 

The executed amount is adding the "solver fee" (this name is used with many different semantics throughout the codebase, but here it's supposed to mean the fee that the solver should use when evaluating the objective value which factors our the 0.002 ETH subsidy we give on market orders), but it should be adding the "user fee" instead.

#2151 should address the issue.

I assume it's ok to report the fully executed buy and sell amounts (after all fees) for the EBBO test.

**Comment by fhenneke** [comment]
> I assume it's ok to report the fully executed buy and sell amounts (after all fees) for the EBBO test.

Yes, this is exactly what we need: the amount users send us (i.e. before deducting any fees in the sell token) and the amount users receive (i.e. after deducting any fees in the buy token).
